.github/workflows/e2e-health-report.yml

```
@@ -44,31 +45,35 @@ jobs:
-              --arg url "$RUN_URL" \
-              '{text: ($text + "\nFull report: <" + $url + "|view run artifacts>")}')"
+          set -euo pipefail
+          [ -z "$REPORTS_DEPLOY_KEY" ] && { echo "::error::REPORTS_DEPLOY_KEY not set"; exit 1; }
+          REPORT_SRC="test/scripts/e2e-health/e2e-health-report.html"
+          [ ! -f "$REPORT_SRC" ] && { echo "::error::Report file not found: $REPORT_SRC"; exit 1; }
+          install -d -m 700 ~/.ssh
+          echo "$REPORTS_DEPLOY_KEY" > ~/.ssh/reports_deploy_key
+          chmod 600 ~/.ssh/reports_deploy_key
+          # Use GitHub's pinned host keys instead of trust-on-first-use ssh-keyscan
+          cat >> ~/.ssh/known_hosts <<'EOF'
+github.com ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCj7ndNxQowgcQnjshcLrqPEiiphnt+VTTvDP6mHBL9j1aNUkY4Ue1gvwnGLVlOhGeYrnZaMgRK6+PKCUXaDbC7qtbW8gIkhL7aGCsOr/C56SoTKygFR7H6H+TuK4P8PCAP2iBNt89f6MkLjGXS8kWw/Ps2V4+S5FBl7GITMHJXoOGLidnxFWGwkOmSe5e36PBo/tUyLwopZe2aiBBlvOhVs2nOE4blnP8qMIxY3L6n6CbMKcgO91PJXJL3HHF6L5WjMuEpocLHO+3t9D2IFLe2rHq3l0Yf16r3CAnH/2vlvG/6RrCNcsBq8Ai4C7zCAAvAAAAADAQAB
```

actionlint (1.7.12)

[error] could not parse as YAML: could not find expected ':'

(syntax-check)